src/java.base/share/classes/java/lang/reflect/Constructor.java

```diff
@@ -656,13 +657,22 @@ public AnnotatedType getAnnotatedReceiverType() {
             return null;
         }
 
+        TypeVariable<?>[] v = enclosingClass.getTypeParameters();
+        Type o = resolveOwner(enclosingClass);
+        Type t;
+        if (o != null || v.length > 0) {
+            t = ParameterizedTypeImpl.make(enclosingClass, v, o);
```

Comment:
I think these should be instantiated by a GenericsFactory. Ideally you shouldn't have to import the reflective object directly.

Reply:
I added it to the existing generics factory but as a static method since the instance values are not really required.